### DISCUSSION TITLE
cannot sign transactions on unsupported network

### DISCUSSION THREAD

**Discussion by veyselozkahraman** [WalletConnect/walletconnect-monorepo#2903] [Developer Support]
Hello,

I am trying to sign a transaction on an avalanche subnet which is not shown under the chains on walletconnect cloud.
I have tried putting the subnet chain id on `optionalChains`, I did that since when I pass the chain id of the subnet on `chains`, WalletConnect gives the `Unsupported chain` error.

I am able to sign transactions on supported chains.

Is there a workaround for this?

**Comment by ignaciosantise** [comment]
Hey, thanks for reporting. More than happy to help however can you please provide us with more detailed reporting?

1) Version of the package you are using
2) Reproducible code via a codeBox
3) Which wallet (platform and version) you used to get to this error
4) Do you get the errors on https://react-web3wallet.vercel.app/
5) How many users do you know are impacted by this

Appreciate your patience and thanks in advance.

**Reply by veyselozkahraman** [reply]
Hello,

Thank you for your answer, here are the details;

These are the packages I use:
        "@walletconnect/ethereum-provider": "^2.8.6",
        "@walletconnect/modal": "^2.5.9",

I am connecting with Metamask mobile to use WalletConnect.

It looks like WalletConnect v2 does not allow signing transactions on unsupported chains. I have just seen something like this 
https://docs.walletconnect.com/2.0/advanced/multichain/chain-list

Should we submit the network details according to the instructions?

**Reply by xzilja** [reply]
When you propose chains, wallet will only approve ones they support (assuming they implemented v2 protocol correctly). If one of the chains you need to work with was not approved you can send a `wallet_addEthereumChain` request with full chain details, if wallet supports this method they will add it, if not it will error. You can view what wallet approved on `provider.signer` object I believe under session namespaces.

**Reply by veyselozkahraman** [reply]
We are developing a dApp, not a wallet and I am connecting my dApp with Metamask on WalletConnect v2 modal. The chain I am trying to send transactions  is not in this list https://docs.walletconnect.com/2.0/advanced/multichain/chain-list. 

Is it correct that WalletConnect v2 does not allow signing transactions on chains that are not on this list?

If it is correct, should I submit a github issue in the way instructed on the chain list  page(https://github.com/WalletConnect/walletconnect-monorepo/issues/new?assignees=&labels=type%3A+new+chain+request&template=new_chain_to_explorer.md&title=) to get the chain I am working on to the allowed list?

**Reply by finessevanes** [reply]
@veyselozkahraman if the chain is EVM, it should still work. If it's non-EVM, create an issue to get the chain added.
